lib/modules/quality_reporting/submit_data_sequence.rb

```diff
@@ -52,6 +52,41 @@ class SubmitDataSequence < SequenceBase
           assert(search_bundle.total.positive?, "Search for a #{r.resourceType} with identifier #{identifier} returned no results")
         end
       end
+
+      test 'Submit Data single resource submission' do
+        metadata do
+          id '02'
+          link 'https://www.hl7.org/fhir/measure-operation-submit-data.html'
+          description 'Submit resources relevant to a measure, and then verify they persist on the server.'
+        end
+
+        assert(!@instance.measure_to_test.nil?, 'No measure selected. You must run the Prerequisite sequences prior to running Reporting Actions sequences.')
+
+        @client.additional_headers = { 'x-api-key': @instance.api_key, 'Authorization': @instance.auth_header } if @instance.api_key && @instance.auth_header
+
+        resources = [get_data_requirements_resources(@instance.data_requirements_queries).sample]
```

Comment:
Why is this pertained as an array if it is a single resource?

Reply:
We also have to add in the measure report to the resources, so the resources variable here needs to be an array anyway, and the `submit_data` function from `measure_operations.rb` already handles an array of resources as well.

Reply:
That makes sense. So it will need to check more than just the single resource when it sees it is persisted.